src/locutus/model/terminology.py

```diff
@@ -649,6 +652,131 @@ def replace_preferred_terminology(self, editor, preferred_terminology):
             print(f"An error occurred while adding preferred terminology: {e}")
             raise
 
+    def get_user_input(self, resource_type, collection_type, id, code, type):

```

Comment:
function would go nicely in the parent class UserInput since it needs to work with all types.